0004-unflushable-subscription-issue-modes.md

```diff
@@ -0,0 +1,102 @@
+
+## Conventions
+- The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in [RFC 2119](http://tools.ietf.org/html/rfc2119).
+
+## Motivation
+
+In order to use PeerAsset tokens for managing subscriptions to premium services, new issue modes are required.
+The issue modes proposed in this document, `UNFLUSHABLE` & `SUBSCRIPTION`, allow clients to validate time bound subscriptions issued by an entity.
+
+Using PeerAssets for subscription tracking allows service providers to check for a client's subscription without the need of accessing a centralized user database.
+This allows a more decentralized service architecture to be implemented by the service provider.
+Or allows service providers to check for a subscription to an external entity they do not control or have access to.
+
+To illustrate the latter, imagine *organization A* that tracks time bound membership (subscriptions) using PeerAssets.
+*Organization A* may hold a user database that links PeerAssets addresses to a real life identity.
+*Organization B* may decide to offer extra services or discounts to members of *organization A*.
+However, *organization A* wishes not to give *organization B* access to their user database in any way to protect the personal information of their members.
+Since *organization A* uses PeerAssets to track membership, members of *organization A* can prove their membership without having to provide any personal details, not even their username.
+
+## Detailed design
+
+In order to create the `SUBSCRIPTION` issue mode, an additional basic issue mode `UNFLUSHABLE` needs to be specified.
+
+### Unflushable mode
+The `UNFLUSHABLE` issue mode invalidates any card transfer transaction except for the card issue transaction.
+Meaning that only the issuing entity is able to change the balance of a specific address.
+To correctly calculate the balance of a PeerAssets address a client should only consider the card transfer transactions originating from the deck owner.
+
+##### bitflags
+As specified in [0001-peerassets-transaction-specification.proto](0001-peerassets-transaction-specification.proto), the `UNFLUSHABLE` issue mode is activated by setting the `0x10` bitflag.
+
+
+##### Implementation
+Upon validation of a card transfer transaction not originating from the asset owner, one should check whether the `UNFLUSHABLE` bit is set.
+If set, the transaction should be marked invalid and discarded for the balance calculation.
+
+### Subscription mode
+The `SUBSCRIPTION` issue mode marks an address holding tokens as subscribed for a limited timeframe.
+This timeframe is defined by the balance of the account and the time at which the first cards of this token are received.
+
+To check validity of a subscription one should take the timestamp of the first received cards and add the address' balance to it in hours.
```

Comment:
It won't be accurate indeed, that's why the time is expressed in hours.
The chain's time drift is capped to two hours.
So subscription should be used in the order of days rather than seconds.

Do you prefer using the balance as days?
It would make sense to me, people wanting higher resolution can still use decimals.